pkg/pub_package_reader/lib/pub_package_reader.dart

```diff
@@ -220,6 +221,14 @@ Iterable<ArchiveIssue> validatePackageName(String name) sync* {
   }
 }
 
+/// Sanity checks for the package's version.
+Iterable<ArchiveIssue> validatePackageVersion(Version version) sync* {
+  if (version.toString().length > 32) {
+    yield ArchiveIssue('Package version must not exceed 32 characters. '
+        '(Please file an issue if you think you have a good reason for a longer version.)');
+  }
```

Comment:
Some stats based on public APIs:
```
No version numbers with more than 40 chars exists

## Version numbers >= 25 chars: 132

## Version numbers >= 27 chars: 7
Package/version with version number >= 27 chars: 
 - ai_awesome_message has "0.2.0-compatible-flutter-v1.15.4-pre.190"
 - d20 has "0.2.0+test-github-actions-1"
 - measurement has "0.0.1-integration-to-master"
 - measurement has "0.0.1-integration-to-master-b"
 - measurement has "0.0.1-integration-to-master-c"
 - measurement has "0.0.1-integration-to-master-d"
 - measurement has "0.0.1-integration-to-master-e"

## Version numbers >= 30 chars: 1
Package/version with version number >= 30 chars: 
 - ai_awesome_message has "0.2.0-compatible-flutter-v1.15.4-pre.190"

## Version numbers >= 32 chars: 1
Package/version with version number >= 32 chars: 
 - ai_awesome_message has "0.2.0-compatible-flutter-v1.15.4-pre.190"
```

Local builds on the Dart SDK uses a version scheme like:
`2.9.0-edge.d7cb9ba973732e2e92789e3b2e12a8ae8fbdf9ee`

Thinking about it, embedding git revision hash in the `-prerelease.version` or `+build.info` is a plausible use-case.
So maybe 64 chars is a better limit?

Reply:
(I'm not sure I would recommend publishing packages that embeds git revisions in the version number, IMO keeping it simple is a good idea)

Reply:
64 it is